src/Ryujinx.Memory/MemoryBlock.cs

```diff
@@ -365,9 +364,11 @@ public WritableRegion GetWritableRegion(ulong offset, int size)
         /// <param name="pointer">Native pointer</param>
         /// <param name="offset">Offset to add</param>
         /// <returns>Native pointer with the added offset</returns>
-        private IntPtr PtrAddr(IntPtr pointer, ulong offset)
+        private static IntPtr PtrAddr(IntPtr pointer, ulong offset)
         {
+#pragma warning disable CA2020 // Prevent behavioral change caused by built-in operators of IntPtr/UIntPtr
             return (IntPtr)(pointer.ToInt64() + (long)offset);
```

Comment:
```suggestion
            return new IntPtr(pointer.ToInt64() + (long)offset);
```

And remove the pragma ?